### PR TITLE
[TTI] getCommonMaskedMemoryOpCost - consistently use getScalarizationOverhead instead of ExtractElement costs for address/mask extraction.

### PR DESCRIPTION
**Pull request by RKSimon**
These aren't unknown extraction indices, we will be extracting every address/mask element in sequence.

### CODE REVIEW

File: llvm/test/Analysis/CostModel/X86/masked-intrinsic-sizelatency.ll [hunk 1]
```diff
@@ -732,192 +732,192 @@ define i32 @masked_store(<1 x i1> %m1, <2 x i1> %m2, <3 x i1> %m3, <4 x i1> %m4,
...
-; SKX-NEXT:  Cost Model: Found an estimated cost of 20 for instruction: %V4I16 = call <4 x i16> @llvm.masked.gather.v4i16.v4p0(<4 x ptr> undef, i32 1, <4 x i1> %m4, <4 x i16> undef)
-; SKX-NEXT:  Cost Model: Found an estimated cost of 323 for instruction: %V64I8 = call <64 x i8> @llvm.masked.gather.v64i8.v64p0(<64 x ptr> undef, i32 1, <64 x i1> %m64, <64 x i8> undef)
-; SKX-NEXT:  Cost Model: Found an estimated cost of 161 for instruction: %V32I8 = call <32 x i8> @llvm.masked.gather.v32i8.v32p0(<32 x ptr> undef, i32 1, <32 x i1> %m32, <32 x i8> undef)
-; SKX-NEXT:  Cost Model: Found an estimated cost of 80 for instruction: %V16I8 = call <16 x i8> @llvm.masked.gather.v16i8.v16p0(<16 x ptr> undef, i32 1, <16 x i1> %m16, <16 x i8> undef)
-; SKX-NEXT:  Cost Model: Found an estimated cost of 40 for instruction: %V8I8 = call <8 x i8> @llvm.masked.gather.v8i8.v8p0(<8 x ptr> undef, i32 1, <8 x i1> %m8, <8 x i8> undef)
+; SKX-NEXT:  Cost Model: Found an estimated cost of 175 for instruction: %V32I16 = call <32 x i16> @llvm.masked.gather.v32i16.v32p0(<32 x ptr> undef, i32 1, <32 x i1> %m32, <32 x i16> undef)
```
[the remark this answers is not in the document]
Comment (RKSimon) [reply llvm/test/Analysis/CostModel/X86/masked-intrinsic-sizelatency.ll:913]: X86 currently drops down to the default (scalarization) implementation for all non-throughput costkinds

**Comment by RKSimon** [comment]
ping?

**Review by alexey-bataev — APPROVED** [review]
LG

**Review by david-arm — APPROVED** [review]
LGTM! I think this looks correct and more accurate logically than before, since the insert/extract costs could be different depending upon the lane. I think if some costs have gone up, then it's up to the target to fix getVectorInstrCost to be more accurate.

**Pull request merged**